src/python/WMCore/MicroService/Unified/RequestInfo.py

```diff
@@ -60,6 +67,14 @@ def __call__(self, reqRecords):
 
         return workflows
 
+    def clearPileupCache(self):
+        """
+        Clears the in-memory pileup cache for every cycle of
+        the MSTransferor module.
+        Cache stores only the total size for secondary datasets
+        """
+        self.cachePileupSize.clear()
```

Comment:
As we quickly talked on the meeting, this is going to be a single thread application. If we do decide to make some of the processing in multi-threading mode, it would still be fine provided we don't change the `self.cachePileupSize` while we have many threads executing, given that each of them would have its own copy of `self.cachePileupSize`.